auto-value-moshi/src/main/java/com/ryanharter/auto/value/moshi/AutoValueMoshiExtension.java

```diff
@@ -377,14 +378,19 @@ public MethodSpec createWriteMethod(TypeName autoValueClassName,
         .addException(IOException.class);
 
     writeMethod.addStatement("$N.beginObject()", writer);
+
+    NameAllocator nameAllocator = new NameAllocator();
```

Comment:
Done. Decided to go with `this.` when accessing an adapter. 
Pushed as a separate commit for an easier review.